launchd+targets/darwin: Escape XML in plists

linnnus opened the pull request:
### Description

https://github.com/NixOS/nixpkgs/pull/356502 introduced an argument escape to lib.generators.toPlist which makes it escape special characters in the XML. This PR makes nix-darwin use it everywhere.

This is a breaking change, as https://github.com/nixos/nixpkgs/pull/356502#issuecomment-2480637115.

### Checklist

<!--

Please go through the following checklist before opening a non-WIP
pull-request.

Also make sure to read the guidelines found at

  https://nix-community.github.io/home-manager/#sec-guidelines

-->

- [ ] Change is backwards compatible.

- [x] Code formatted with `nix fmt` or
    `nix-shell -p treefmt nixfmt-rfc-style deadnix keep-sorted --run treefmt`.

- [x] Code tested through `nix-shell --pure tests -A run.all`
    or `nix build --reference-lock-file flake.lock ./tests#test-all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/a51598236f23c89e59ee77eb8e0614358b0e896c/modules/programs/lesspipe.nix#L11).

#### Maintainer CC

<!--
If you are updating a module, please @ people who are in its `meta.maintainers` list.
If in doubt, check `git blame` for whoever last touched something.
-->

@midchildan
@khaneliman

khaneliman commented:
starting to see the eval warnings, was there any progress made on this in nixpkgs/nix-darwin ?

linnnus commented:
We [ended up deciding](https://matrix.to/#/!MMDPZwBJsBXZCsLkuj:matrix.org/$iGsiAQpTDgfewpFSB7En60gwnchSRV3pQHW_4cnCKz0?via=nixos.org&via=matrix.org&via=catgirl.cloud) that a complicated solution wasn't worth it compared to the relatively minor breakage. The most common breakage was people doing this:

```nix
ProgramArguments = [
  "/bin/sh"
  "-c"
  "wait4path /nix/store &amp;&amp; /nix/store/...
];
```

So I added a warning to that option specifically. I have updated this PR to do the same.

Apologies for the late answer!

khaneliman commented:
can you rebase and squash

khaneliman commented:
looks like it needs a `nix fmt`

khaneliman commented:
Looks like an eval error: https://buildbot.nix-community.org/#/builders/6063/builds/179/steps/1/logs/nix_error

khaneliman commented:
Buildbot darwin runners seem stuck today, gonna run test suite locally on my macbook before merging.
